lib/nrf5/common/saadc_common.c

```diff
@@ -0,0 +1,389 @@
+buffer address and size should be configured before start ADC.
+The RESULT.PTR register is double-buffered and it can be updated and prepared 
+for the next START task immediately.
+
+@param[in] *ptr unsigned 8 bit. Address pointer of Result buffer
+@param[in] maxcnt unsigned 16 bit. Maximum number of buffer words to transfer
+*/
+
+void saadc_result_buffer(uint32_t *ptr, uint16_t maxcnt)
+{
+	SAADC_RESULT.PTR = (uint32_t) ptr;
+	SAADC_RESULT.MAXCNT = (uint32_t) maxcnt;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief SAADC Enable ADC.
+
+Enable ADC
+*/
+
+void saadc_enable(void)
+{
+	SAADC_ENABLE = SAADC_ENABLE_ENABLE_ENABLED;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief SAADC Disable ADC.
+
+Disable ADC
+*/
+
+void saadc_disable(void)
+{
+	SAADC_ENABLE = SAADC_ENABLE_ENABLE_DISABLED;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief SAADC Start sampling.
+
+Start sampling process and block until the ADC has filled uo the Result buffer
+*/
+
+void saadc_sample_start(void)
+{
+	SAADC_TASKS_SAMPLE = 1;
+	while(SAADC_EVENTS_END == 0);
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief SAADC Start ADC.
+
+Start ADC and block until the ADC has started
+*/
+
+void saadc_start(void)
+{
+	SAADC_TASKS_START = 1;
+	while(SAADC_EVENTS_STARTED == 0);
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief SAADC Stop ADC.
+
+Stop ADC and block until the ADC has stopped
+*/
+
+void saadc_stop(void)
+{
+	SAADC_TASKS_STOP = 1;
+	while(SAADC_EVENTS_STOPPED == 0);
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief SAADC High limit status.
+
+Return 1 if the result reached high limit
+
+@param[in] channel unsigned 8 bit. ADC channel selection
+@returns unsigned 32 bit data word.
+*/
+
+uint32_t saadc_limit_high(uint8_t channel)
+{
+	return SAADC_EVENTS_CH_LIMITH[channel];
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief SAADC Low limit status.
+
+Return 1 if the result reached low limit
+
+@param[in] channel unsigned 8 bit. ADC channel selection
+@returns unsigned 32 bit data word.
+*/
+
+uint32_t saadc_limit_low(uint8_t channel)
+{
+	return SAADC_EVENTS_CH_LIMITL[channel];
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief SAADC Reset limit events.
+
+Clear high/low limit event
+
+@param[in] channel unsigned 8 bit. ADC channel selection
+*/
+
+void saadc_limit_reset(uint8_t channel)
+{
+	SAADC_EVENTS_CH_LIMITH[channel] = 0;
+	SAADC_EVENTS_CH_LIMITL[channel] = 0;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief SAADC Sampling process status.
+
+Return 1 if the sampling process is done and the ADC has filled up the Result
+buffer.
```

Comment:
and what if it hasn't?  Is this just a bool?

Reply:
The EVENTS_END (and other events) is a register so I prefer to leave the return value as same as defined type, otherwise might got stylecheck.